x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_activity_button.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiText, EuiTourStep } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { useStartServices } from '../../../../hooks';
+
+export const AgentActivityButton: React.FC<{
+  onClickAgentActivity: () => void;
+  showAgentActivityTour: { isOpen: boolean };
+}> = ({ onClickAgentActivity, showAgentActivityTour }) => {
+  const { uiSettings } = useStartServices();
+
+  const [agentActivityTourState, setAgentActivityTourState] = useState(showAgentActivityTour);
+
+  const hideTour = () => uiSettings.get('hideAgentActivityTour', false);
+
+  const setHideTour = () => uiSettings.set('hideAgentActivityTour', true);
+
+  useEffect(() => {
+    setAgentActivityTourState(showAgentActivityTour);
+  }, [showAgentActivityTour, setAgentActivityTourState]);
+
+  const button = (
```

Comment:
(nit) This seems small enough that it could be written as inline JSX, especially since we don't need to render it in multiple places as far as I can tell.

Reply:
It is used in 2 places (with and without tour), that is why I moved it to a variable.